cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+						kubeDeploy = true
+					} else {
+						die("wr-manager deployment found but not healthy: %s", err)
+					}
+				}
+			}
+		}
+
+		if !kubeDeploy {
+			info("found existing healthy wr-manager deployment, reconnecting")
+		} else {
+			info("please wait while wr is deployed to the cluster")
+		}
+
+		// Daemonise
+		fwPidPath := filepath.Join(config.ManagerDir, "kubernetes_resources.fw.pid")
+		umask := 007
+		child, context := daemonize(fwPidPath, umask, extraArgs...)
+		if child != nil {
+			// PostParent() (Runs in the parent process after spawning child)
+			jq = connect(120*time.Second, true)
+			if jq == nil {
+				die("could not talk to wr manager after 120s")
+			}
+
+			// The remote manager is running, read the resource file to determine the name
+			// of the pod to fetch the client.token from.
+
+			// Read the manager pod's name from resource file
+			file, err := os.Open(resourcePath)
+			if err != nil {
+				die("Could not open resource file with path: %s", err)
+			}
+			decoder := gob.NewDecoder(file)
+			err = decoder.Decode(resources)
+			if err != nil {
+				die("Error decoding resource file: %s", err)
+			}
+			managerPodName := resources.Details["manager-pod"]
+			namespace := resources.Details["namespace"]
+
+			internal.LogClose(appLogger, file, "resource file", "path", resourcePath)
+
+			// cat the contents of the client.token in the running manager, so we can
+			// write them to disk locally, and provide the URL for accessing the web interface
+			stdOut, _, err := c.Client.ExecInPod(managerPodName, "wr-manager", namespace, []string{"cat", podBinDir + ".wr_" + config.Deployment + "/client.token"})
+
+			if err != nil {
+				die("something went executing the command to retrieve the token: %s", err)
+			}
+			token := stdOut
+
+			// Write token to file
+			err = ioutil.WriteFile(config.ManagerTokenFile, []byte(token), 0644)
+			if err != nil {
+				warn("Failed to write token to file: %s", err)
+			}
+			info("wr manager remotely started in pod %s (%s)", managerPodName, sAddr(jq.ServerInfo))
+			info("wr's web interface can be reached locally at https://%s:%s/?token=%s", jq.ServerInfo.Host, jq.ServerInfo.WebPort, token)
+		} else {
+			// daemonized child, that will run until signalled to stop
+			// Set up logging to file
+			kubeLogFile := filepath.Join(config.ManagerDir, kubeLogFileName)
+			fh, err := log15.FileHandler(kubeLogFile, log15.LogfmtFormat())
+			if err != nil {
+				warn("wr manager could not log to %s: %s", kubeLogFile, err)
+			} else {
+				l15h.AddHandler(appLogger, fh)
+			}
+
+			defer func() {
+				err := context.Release()
+				if err != nil {
+					warn("daemon release failed: %s", err)
+				}
+			}()
+			info("In daemon")
+
+			debugStr := ""
+			if cloudDebug {
+				debugStr = " --debug"
+			}
+
+			// Look for a set of resources in the manager directory
+			// If found, load them else use a new empty set.
+			info("Checking resources")
```

Comment:
likewise, this and some (but not all) subsequent info lines should probably be removed; only keep the ones that someone might gain some insight from when reading the log file